src/sparse/impl/KokkosSparse_spmv_tensor_core_impl.hpp

```diff
@@ -0,0 +1,373 @@
+                  int ti = fi / 16;
+                  int tj = fi % 16;
+                  sx(wx, ti, tj) = half(x(j * A.blockDim() + ti, yj+tj));
+               }
+#ifdef __CUDA_ARCH__
+               load_matrix_sync(fx, &sx(wx,0,0), 16);
+#endif
+            }
+
+            int off = ci - A.graph.row_map(i); // which block in this row
+            AScalar *ap = rowView.local_row_in_block(off, 0); // offset of this block
+            // BlockCrsMatrix stores it's values in the normal Crs form.
+            // therefore, the pitch of each row is different, since it depends
+            // on how many blocks are in that row.
+            // the pitch between rows within each row of blocks is 
+            // blockDim * the length of the row (in blocks)
+            unsigned lda = A.blockDim() * rowView.length;
+            // team loads A fragment
+               for (int fi = mbr.team_rank(); fi < 16*16; fi += mbr.team_size()) {
+                  int ti = fi / 16;
+                  int tj = fi % 16;
+                  using MulTy = typename wider<AScalar,YScalar>::type;
+                  sa(ti, tj) = half(MulTy(ap[(ti * lda) + tj]) * MulTy(alpha));
+               }
+
+            mbr.team_barrier(); // ensure that warp 0 has loaded A fragment into scratch memory
+            if (yj < y.extent(1)) {
+#ifdef __CUDA_ARCH__
+               load_matrix_sync(fa, (half*)&sa(0,0), 16); // load fa from shared memory
+#endif
+            }
+            mbr.team_barrier(); // prevent warp 0 from loading the next A fragment until everyone has used it
+
+            // fy = fa * fx + fy
+            if (yj < y.extent(1)) {
+#ifdef __CUDA_ARCH__
+               mma_sync(fy, fa, fx, fy);
+#endif
+            } 
+         }
+
+         // store y. The store is staged through shared memory to convert back to 
+         // the YMatrix scalar type
+         if (yj < y.extent(1)) {
+#ifdef __CUDA_ARCH__
+            // store_matrix_sync(yp, fy, ldy, yLayout);
+            store_matrix_sync(&sy(wx,0,0), fy, 16, nvcuda::wmma::mem_row_major);
+#endif
+            // warp collaborative store of Y fragment to destination matrix
+            for (int fi = lx; fi < 16*16; fi += THREADS_PER_WARP) {
+               int ti = fi / 16;
+               int tj = fi % 16;
+               y(yi + ti, yj+tj) = YScalar(sy(wx, ti, tj));
+            }
+         }
+      } // yj
+   }
+};
+
+/* y = beta * y + alpha * A * x
+
+   General tensor core implementation of y = beta * y + alpha * A * x
+   alpha_beta_n_tc: do alpha/beta, mode: N, with tensor cores
+*/
+template <
+   typename AMatrix,
+   typename XMatrix,
+   typename YMatrix
+>
+void spmv_alpha_beta_n_tc(
+typename YMatrix::value_type alpha,
+const AMatrix& A,
+const XMatrix& x,
+typename YMatrix::value_type beta,
+const YMatrix& y
+) {
+   if (0 != y.extent(1) % A.blockDim()) {
+      Kokkos::Impl::throw_runtime_exception("y.extent(1) not a multiple of A block dimension");
```

Comment:
Couldn't we just pad the multivector with additional vectors full of zeros when loading the y fragments?